### PR TITLE
Add a new generic type in order to use the Cursor type in Android

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR brings generics to the driver.

The main changes are:
- Classes like `Connection`, `Database` and so on have the generic type
- `AndroidCursor` extends `Cursor` from android API
- Renamed `getType` method in `AndroidCursor` to avoid conflictive members with `Cursor`

Please @juanpedromoreno could you review? Thanks

### CODE REVIEW

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-import android.database.Cursor
+import android.content.ContentResolver
+import android.database.{ContentObserver, CharArrayBuffer, DataSetObserver, Cursor}
+import android.net.Uri
+import android.os.Bundle
 import com.fortysevendeg.mvessel.api.CursorType.Value
 import com.fortysevendeg.mvessel.api.{CursorType, CursorProxy}
 
-class AndroidCursor(cursor: Cursor) extends CursorProxy {
+class AndroidCursor(cursor: Cursor) extends CursorProxy with Cursor {
```
Comment (47degci) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/965563501/issues/source?bid=2442814&fileBranchId=2859700#l10)

File: core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala [hunk 2]
```diff
@@ -13,9 +14,9 @@ import com.fortysevendeg.mvessel.util.StreamUtils._
 
 import scala.util.{Failure, Success, Try}
 
-class PreparedStatement(
+class PreparedStatement[T <: CursorProxy](
```
Comment (47degci) [inline core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/965563529/issues/source?bid=2442814&fileBranchId=2859700#l17)

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
 
-  override def getType(columnIndex: Int): Value = cursor.getType(columnIndex) match {
+  override def getCursorType(columnIndex: Int): Value = cursor.getType(columnIndex) match {
     case Cursor.FIELD_TYPE_NULL => CursorType.Null
         case Cursor.FIELD_TYPE_INTEGER => CursorType.Integer
```
Comment (juanpedromoreno) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala]: Seems like a bad indentation.

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala [hunk 4]
```diff
@@ -63,4 +66,32 @@ class AndroidCursor(cursor: Cursor) extends CursorProxy {
...
+
+  override def getWantsAllOnMoveCalls: Boolean = cursor.getWantsAllOnMoveCalls
+
+  override def registerContentObserver(observer: ContentObserver): Unit = cursor.registerContentObserver(observer)
+
+  override def unregisterContentObserver(observer: ContentObserver): Unit = cursor.unregisterContentObserver(observer)
+
+  override def registerDataSetObserver(observer: DataSetObserver): Unit = cursor.registerDataSetObserver(observer)
+
+  override def unregisterDataSetObserver(observer: DataSetObserver): Unit = cursor.unregisterDataSetObserver(observer)
```
Comment (juanpedromoreno) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala]: has been the line limit reached?

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala [hunk 5]
```diff
@@ -63,4 +66,32 @@ class AndroidCursor(cursor: Cursor) extends CursorProxy {
...
+
+  override def deactivate(): Unit = cursor.deactivate()
+
+  override def copyStringToBuffer(columnIndex: Int, buffer: CharArrayBuffer): Unit = cursor.copyStringToBuffer(columnIndex, buffer)
```
Comment (juanpedromoreno) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala]: has been the line limit reached?

**Comment by juanpedromoreno** [comment]
Minor code style comments. LGTM, codacy and coveralls are red though.

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala [hunk 6]
```diff
@@ -1,50 +1,118 @@
...
+import android.content.ContentResolver
+import android.database.{CharArrayBuffer, ContentObserver, Cursor, DataSetObserver}
+import android.net.Uri
+import android.os.Bundle
 import com.fortysevendeg.mvessel.api.CursorType.Value
-import com.fortysevendeg.mvessel.api.{CursorType, CursorProxy}
+import com.fortysevendeg.mvessel.api.{CursorProxy, CursorType}
 
-class AndroidCursor(cursor: Cursor) extends CursorProxy {
+class AndroidCursor(val cursor: Cursor)
```
Comment (47degci) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/970112756/issues/source?bid=2442814&fileBranchId=2859700#l10)

File: core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala (hunk 2 above)
Comment (47degci) [inline core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/970112784/issues/source?bid=2442814&fileBranchId=2859700#l17)

File: android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala (hunk 6 above)
Comment (47degci) [inline android-driver/src/main/scala/com/fortysevendeg/mvessel/api/impl/AndroidCursor.scala:10]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/970159385/issues/source?bid=2442814&fileBranchId=2859700#l10)

File: core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala (hunk 2 above)
Comment (47degci) [inline core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala:17]: Codacy Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/970159414/issues/source?bid=2442814&fileBranchId=2859700#l17)

**Pull request merged**